### PR TITLE
Modify hardhat config network url

### PR DESCRIPTION
**Pull request by hswopeams**
to be read from .env file

Previously, the Infura url was hardcoded in the hardhat config file and only the API key was read from the .env file. This change makes it possible to use any node URL without being limited to Infura.

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ import { HardhatUserConfig } from "hardhat/config";
 
 const { task } = require("hardhat/config");
 const testMnemonic = 'inhale wood champion certain immense wash pepper enact enrich infant purse maid'
-const INFURA_KEY = process.env.INFURA_API_KEY;
+const RINKEBY_URL = process.env.RINKEBY_URL;
+const ROPSTEN_URL = process.env.ROPSTEN_URL;
+const MAINNET_URL = process.env.MAINNET_URL;
```
Comment (bidyut-arianelabs) [inline hardhat.config.ts:17]: These 3 Environment variables should be added in the `ci.yaml`, `code-coverage-check.yaml`.
Without these, the CI will break
e.g. 
```
jobs:
  build:
    env:
      RINKEBY_URL: https://provider_url/project_key
      ROPSTEN_URL: https://provider_url/project_key
      MAINNET_URL: https://provider_url/project_key
```

**Review by bidyut-arianelabs — CHANGES_REQUESTED** [review]

File: hardhat.config.ts (hunk 1 above)
Comment (hswopeams) [reply hardhat.config.ts:17]: Yes, I know. Working on it. I can't add it until a secret is created.

**Review by cliffhall — APPROVED** [review]
Assuming the secrets get added and the build passes, LGTM! 👍

**Review by ghost — DISMISSED** [review]

**Review by cliffhall — DISMISSED** [review]
LGTM! 👍

**Review by zajck — APPROVED** [review]

File: hardhat.config.ts (hunk 1 above)
Comment (bidyut-arianelabs) [reply hardhat.config.ts:17]: Resolving this conversation as the Github actions are working fine now.

**Review by hswopeams — COMMENTED** [review]
LGTM

**Review by bidyut-arianelabs — APPROVED** [review]

**Pull request merged**